microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/project/JDTMicroProfileProject.java

```diff
@@ -119,4 +107,12 @@ public Integer getPropertyAsInteger(String key, Integer defaultValue) {
 				.collect(Collectors.toList());
 	}
 
+	private List<IConfigSource> getConfigSources() {
+		TreeSet<IConfigSource> configSourceSet = new TreeSet<>((a, b) -> b.getOrdinal() - a.getOrdinal());

```

Comment:
I think it can be a good idea to cache the list because getConfigSources will be consumeda lot of time.

Reply:
The reason I am not caching the sources is because for https://github.com/redhat-developer/vscode-quarkus/issues/371 I need to reload the list of configuration sources when a new `application-*.properties` file is added/deleted. Do you have an idea for a strategy to cache these files?

Reply:
Maybe I could make a `static` config source provider and a `dynamic` config source provider, and only reload the dynamic config sources. Does this sound like a good idea?

Reply:
For the moment we will skip the cache because it's fast.

Reply:
>Do you have an idea for a strategy to cache these files?

We must implement implement IResourceChangeListener if I remember, but let's skip cache for the moment.